### PR TITLE
build.yml: Remove -G from cibuild.sh command line

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
Signed-off-by: Xiang Xiao <xiaoxiang@xiaomi.com>

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
work with https://github.com/apache/incubator-nuttx/pull/841 to find the residuum of distclean/clean

**Comment by btashton** [comment]
I think the point here was to improve the speed of the build by using git instead? Or are we just wanting to do this to test the distclean?

**Comment by patacongo** [comment]
> I think the point here was to improve the speed of the build by using git instead? Or are we just wanting to do this to test the distclean?

Should this be merged?  You asked a question that has not been answered.  Note that I have already merged #841 which is referenced by this PR.  Is there any dependency?

**Comment by xiaoxiang781216** [comment]
> > I think the point here was to improve the speed of the build by using git instead? Or are we just wanting to do this to test the distclean?
> 
> Should this be merged? You asked a question that has not been answered. Note that I have already merged #841 which is referenced by this PR. Is there any dependency?

#841 add the distclean/clean check but not enable, This PR enable that check, so it isn't a big issue that we just merge #841,

**Comment by xiaoxiang781216** [comment]
> I think the point here was to improve the speed of the build by using git instead? Or are we just wanting to do this to test the distclean?

Yes, git clean can improve the speed a little bit, but the correction is more important than the speed. Most people never check whether the repo is in the clean state after "make distclean". This patch can ensure developer remember it.

**Comment by btashton** [comment]
Ok. I'm just worried that we are going to hit an issue where our CI is taking too long. So trying to make sure the little things that are in every build don't add up. 

In this case I guess it's probably worth the extra now that we are checking that distclean works correctly.

**Comment by btashton** [comment]
Any idea why the build failed. It seems unrelated.

**Comment by xiaoxiang781216** [comment]
> Any idea why the build failed. It seems unrelated.

I will rebase and check again. It suppose more error will find after  #841.

File: tools/testbuild.sh [hunk 1]
```diff
@@ -234,6 +234,10 @@ function build {
     fail=1
   fi
 
+  # Ensure HOST config restore after refresh
+
+  ./tools/sethost.sh
```
Comment (xiaoxiang781216) [inline tools/testbuild.sh:239]: Let's move into tools/refresh.sh

File: tools/refresh.sh [hunk 2]
```diff
@@ -33,6 +33,7 @@
 #
 
 WD=`test -d ${0%/*} && cd ${0%/*}; pwd`
+TOPDIR="${WD}/.."
```
Comment (xiaoxiang781216) [inline tools/refresh.sh:36]: don't need, could use: ./tools/sethost.sh

File: tools/refresh.sh [hunk 3]
```diff
@@ -285,15 +286,17 @@ for CONFIG in ${CONFIGS}; do
...
 
+  if [ -e SAVEMake.defs ]; then
+    mv SAVEMake.defs Make.defs || \
+      { echo "ERROR: Failed to move SAVEMake.defs to Make.defs"; exit 1; }
+  fi
+
   if [ -e SAVEconfig ]; then
     mv SAVEconfig .config || \
       { echo "ERROR: Failed to move SAVEconfig to .config"; exit 1; }
+    ${TOPDIR}/tools/sethost.sh 1>/dev/null
```
Comment (xiaoxiang781216) [inline tools/refresh.sh:297]: Should we change like other place:
 if [ "X${debug}" = "Xy" ]; then
   tools/sethost.sh V=1
 else
   tools/sethost.sh 1>/dev/null
  fi

**Comment by liuguo09** [comment]
https://github.com/apache/incubator-nuttx-apps/pull/213 fixes the sim:unionfs check build failure. Once merged, re-run this PR check build.

File: libs/libnx/nxfonts/Makefile.sources [hunk 4]
```diff
@@ -325,3 +325,4 @@ clean:
 distclean: clean
 	$(call DELFILE, nxfonts_convert_*bpp.c)
 	$(call DELFILE, nxfonts_bitmaps_*.c)
+	$(call DELFILE, nxfonts_tom-thumb-4x6.c)
```
Comment (xiaoxiang781216) [inline libs/libnx/nxfonts/Makefile.sources:328]: let's change the generated file to  nxfonts_bitmaps_tom-thumb-4x6.c like other font?

Comment (liuguo09) [reply libs/libnx/nxfonts/Makefile.sources]: ok, done.

**Comment by xiaoxiang781216** [comment]
LGMT, can any body help review the chagne and merge?

**Comment by xiaoxiang781216** [comment]
Does anyone have any feedback? I want to enable the more check before the end of day.

**Comment by btashton** [comment]
@xiaoxiang781216 Looks like this added almost 15min to the build times for macOS.  Do you think there are improvements we can make to distclean to help with the performance?  If it is IO bound maybe building on a RAM disk would help? https://gist.github.com/htr3n/344f06ba2bb20b1056d7d5570fe7f596

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 Looks like this added almost 15min to the build times for macOS. Do you think there are improvements we can make to distclean to help with the performance? If it is IO bound maybe building on a RAM disk would help? https://gist.github.com/htr3n/344f06ba2bb20b1056d7d5570fe7f596

Yes, we can try both RAM disk and ccache to improve the performance.

**Pull request merged**